saleor/graphql/order/mutations/fulfillments.py

```diff
@@ -158,6 +158,23 @@ def check_lines_for_duplicates(cls, lines_ids):
                 }
             )
 
+    @classmethod
+    def check_lines_for_preorder(cls, order_lines):
```

Comment:
While this PR was in the works we added the ability to create "waiting" fulfillments. These are fulfillment objects that are treated like "stubs" - they exist in the DB but they do not decrease stocks. Admin can approve these waiting fulfillments which then triggers all the logic.

To support preorders with waiting fulfillments, we should add `check_lines_for_preorder` and any other logic changed in this mutation to `FulfillmentApprove` mutation.

Reply:
@maarcingebala I've changed it as you asked (I hope), but I'm not so convinced to this solution, I would be glad if you could double-check if it's ok.

Reply:
I think it looks OK, we need to yet test it in QA.